libobs/CMakeLists.txt

```diff
@@ -135,357 +204,108 @@ target_sources(
-    libobs PROPERTIES SOVERSION "1" BUILD_RPATH
-                                    "$<TARGET_FILE_DIR:OBS::libobs-opengl>")
-
-elseif(OS_POSIX)
-  if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-    target_compile_definitions(libobs PRIVATE ENABLE_DARRAY_TYPE_TEST)
-  endif()
-
-  find_package(LibUUID REQUIRED)
-  find_package(X11 REQUIRED)
-  find_package(
-    XCB
-    COMPONENTS XCB
-    OPTIONAL_COMPONENTS XINPUT
-    QUIET)
-  find_package(X11_XCB REQUIRED)
-
-  target_sources(
-    libobs
-    PRIVATE obs-nix.c
-            obs-nix-platform.c
-            obs-nix-platform.h
-            obs-nix-x11.c
-            util/threading-posix.c
-            util/threading-posix.h
-            util/pipe-posix.c
-            util/platform-nix.c)
-
-  target_link_libraries(libobs PRIVATE X11::X11_xcb XCB::XCB LibUUID::LibUUID)
-
-  if(USE_XDG)
-    target_compile_definitions(libobs PRIVATE USE_XDG)
-  endif()
-
-  if(ENABLE_PULSEAUDIO)
-    find_package(PulseAudio REQUIRED)
-    obs_status(STATUS "-> PulseAudio found - audio monitoring enabled")
-    target_sources(
-      libobs
-      PRIVATE audio-monitoring/pulse/pulseaudio-output.c
-              audio-monitoring/pulse/pulseaudio-enum-devices.c
-              audio-monitoring/pulse/pulseaudio-wrapper.c
-              audio-monitoring/pulse/pulseaudio-wrapper.h
-              audio-monitoring/pulse/pulseaudio-monitoring-available.c)
-
-    target_link_libraries(libobs PRIVATE ${PULSEAUDIO_LIBRARY})
-  else()
-    obs_status(WARNING "-> No audio backend found - audio monitoring disabled")
-    target_sources(libobs PRIVATE audio-monitoring/null/null-audio-monitoring.c)
-  endif()
-
-  find_package(Gio)
-  if(TARGET GIO::GIO)
-    target_link_libraries(libobs PRIVATE GIO::GIO)
-
-    target_sources(libobs PRIVATE util/platform-nix-dbus.c
-                                  util/platform-nix-portal.c)
-  endif()
-
-  if(TARGET XCB::XINPUT)
-    target_link_libraries(libobs PRIVATE XCB::XINPUT)
-  endif()
-
-  if(ENABLE_WAYLAND)
-    find_package(
-      Wayland
-      COMPONENTS Client
-      REQUIRED)
-    find_package(Xkbcommon REQUIRED)
-
-    target_link_libraries(libobs PRIVATE Wayland::Client Xkbcommon::Xkbcommon)
-
-    target_sources(libobs PRIVATE obs-nix-wayland.c)
-  endif()
-
-  if(OS_LINUX)
-    target_link_libraries(glad PRIVATE ${CMAKE_DL_LIBS})
-  endif()
+configure_file(obsconfig.h.in "${CMAKE_BINARY_DIR}/config/obsconfig.h" @ONLY)
+target_include_directories(libobs PUBLIC "$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/config>"
+                                         "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>")
+target_compile_definitions(libobs PUBLIC HAVE_OBSCONFIG_H)
 
-  if(OS_FREEBSD)
-    find_package(Sysinfo REQUIRED)
-    target_link_libraries(libobs PRIVATE Sysinfo::Sysinfo)
-  endif()
+list(
+  APPEND
+  obs.h
+  obs-config.h
+  obs-defs.h
+  obs-data.h
+  obs-interaction.h
+  obs-internal.h
+  obs-module.h
+  obs-properties.h
+  obs-ui.h
+  public_headers
```

Comment:
Shouldn't `public_headers` be the first item after `APPEND`? Should it be a different case to differentiate it at a glance?